src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -291,92 +286,103 @@ object DataModelInstanceRelation {
       case a => throw new CdfSparkException(unknownPropertyTypeMessage(a))
     }
 
-  private def jsonFromDouble(num: Double): Json =
-    Json.fromDouble(num).getOrElse(throw new CdfSparkException(s"Numeric value $num"))
-  private def tryGetValue(propertyType: String, nullable: Boolean): PartialFunction[Any, Json] = // scalastyle:off
+  private def toPropertyType(propertyType: String): PartialFunction[Any, PropertyType] = // scalastyle:off
     propertyType.toLowerCase match {
-      case "float32" | "float64" | "numeric" => {
-        case null if !nullable => throw new CdfSparkException(propertyNotNullableMessage(propertyType))
-        case null => Json.Null // scalastyle:off null
-        case x: Double => jsonFromDouble(x)
-        case x: Int => jsonFromDouble(x.toDouble)
-        case x: Float => jsonFromDouble(x.toDouble)
-        case x: Long => jsonFromDouble(x.toDouble)
-        case x: java.math.BigDecimal => jsonFromDouble(x.doubleValue)
-        case x: java.math.BigInteger => jsonFromDouble(x.doubleValue)
+      case "float32" => {
+        case x: Float => Float32Property(x)
+        case x: Double => Float32Property(x.toFloat)
+        case x: Int => Float32Property(x.toFloat)
+        case x: Long => Float32Property(x.toFloat)
+        case x: java.math.BigDecimal => Float32Property(x.floatValue())
+        case x: java.math.BigInteger => Float32Property(x.floatValue())
+        case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, propertyType))
```

Comment:
I think handling each case in a separate function will reduce the cyclomatic complexity and make the code easier to read hehe 😁